components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -322,11 +321,75 @@ sealed class ContentAction : BrowserAction() {
     /**
      * Updates permissions highlights of the [ContentState] with the given [sessionId].
      */
-    data class UpdatePermissionHighlightsStateAction(
-        val sessionId: String,
-        val highlights: PermissionHighlightsState
-    ) : ContentAction()
-
+    sealed class UpdatePermissionHighlightsStateAction : ContentAction() {
+        /**
+         * Updates the notificationChanged property of the [PermissionHighlightsState] with the given [sessionId].
+         */
+        data class NotificationChangedAction(val sessionId: String, val value: Boolean) :
```

Comment:
nit: For new actions I started to use `tabId` since a "tab" is easier to understand than a "session".